src/mjolnir/ferry_connections.cc

```diff
@@ -232,7 +232,7 @@ std::pair<uint32_t, bool> ShortestPath(const uint32_t start_node_idx,
           sequence<Edge>::iterator element = edges[edge.second];
           auto update_edge = *element;
           if (update_edge.attributes.importance > kFerryUpClass) {
-            update_edge.attributes.importance = kFerryUpClass;
+            update_edge.attributes.importance_hierarchy = kFerryUpClass;
```

Comment:
this is the one of the most relevant changes: don't change the `importance` but remember the important relevant for hierarchies in a new attribute